packages/node/src/indexer/blockDispatcher/worker-block-dispatcher.service.ts

```diff
@@ -158,6 +167,20 @@ export class WorkerBlockDispatcherService
        * worker2: 4,5,6
        */
       const workerIdx = this.getNextWorkerIndex();
+      const batchSize = Math.min(cleanedBlocks.length, await this.maxBatchSize(workerIdx));
+
+      logger.info(`smart batch size: ${batchSize}`)
+
+      if(batchSize === 0) {
+        return this.enqueueBlocks(cleanedBlocks);
+      }
+
+      if(batchSize < cleanedBlocks.length) {
+        cleanedBlocks.slice(0, batchSize).map((height) => this.enqueueBlock(height, workerIdx));
+        this.latestBufferedHeight = cleanedBlocks[batchSize-1];
+        return this.enqueueBlocks(cleanedBlocks.slice(batchSize));
+      }
```

Comment:
Here we're discarding valuable data and will end up duplicating dictionary queries.

Instead I think we could update getNextWorkerIndex (or some other function) to get a worker with available memory